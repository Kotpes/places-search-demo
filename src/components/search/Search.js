//@flow
import React from "react";
import css from './Search.module.css'

type Props = {
  onChange: Function
};

const Search = ({ onChange }: Props) => {
  return (
    <section className={css.component}>
      <input 
        type="text" 
        placeholder="Start typing" 
        onChange={(element: SyntheticInputEvent<HTMLInputElement>) => onChange(element.target.value)} />
    </section>
  );
};

export default Search;
