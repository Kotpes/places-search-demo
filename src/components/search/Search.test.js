import React from 'react'
import TestRenderer from 'react-test-renderer'
import Search from './Search'

it(`Matches the snapshot`, () => {
  const props = {
    onChange: jest.fn()
  }
  const component = TestRenderer.create(<Search {...props} />)  
  expect(component.toJSON()).toMatchSnapshot()
})