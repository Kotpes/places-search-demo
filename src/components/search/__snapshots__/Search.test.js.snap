// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1`] = `
<section
  className="component"
>
  <input
    onChange={[Function]}
    placeholder="Start typing"
    type="text"
  />
</section>
`;
